src/xhr.js

```diff
@@ -302,8 +334,15 @@ function createXhrRequest(method, url, init) {
  * @return {!FetchResponse}
  */
 function assertSuccess(response) {
-  user.assert(response.status >= 200 && response.status < 300,
-      'HTTP error %s', response.status);
+  if (!(response.status >= 200 && response.status < 300)) {
+    const err = user.createError('HTTP error %s', response.status);
+    // If 415 or in the 5xx range, mark it as retriable.
+    if (response.status == 415 ||
+        (response.status >= 500 && response.status < 600)) {
+            err.retriable = true;
```

Comment:
Indent messed up. Also, this would be nice to put in a standalone function `isRetriable` or such.